### ISSUE TITLE
VPE does not reuse the same instance of the same BP

### ISSUE THREAD

**Issue by cmungall** [geneontology/noctua-visual-pathway-editor#22]
To recreate:

 1. create two activity units A1 A2
 2. assign the same BP class to both units

E.g

<img width="740" alt="image" src="https://user-images.githubusercontent.com/50745/164122457-9f412867-cae9-4820-a856-fe24749c579b.png">

In the underlying RDF, this renders as two disconnected BPs, also in the GE:

<img width="783" alt="image" src="https://user-images.githubusercontent.com/50745/164122533-68d4493f-6891-405a-8a41-b0f56f31550d.png">

(please ignore the obviously incorrect biology I picked the first thing in autocomplete)

This is not necessarily incorrect behavior, but it does result in different structures from what we get from either editing in the GE or via the biopax2gocam conversion.

If we are saying that there is no significance to whether the BP instances are shared or whether we make duplicate instances, then we need to explicitly document this in the shex, otherwise some applications will make incorrect assumptions and give unusual or incorrect results.

I would personally prefer to see the BP instance shared. This is more accurate biologically and allows for better QC (for example: checking that all steps in a pathwat have been fully described). Note the assumption that shared BP class ID == same instance does not always hold. Consider a large pathway model with two MAPK subcomponents that are distinct. It would be hard to figure out heuristically after the fact which activity goes with which MAPK instance.

**Comment by vanaukenk** [comment]
@cmungall 
I agree and note that this is a more general issue that we also have with the graph editor and with nodes besides BP, e.g. CC and anatomy terms.  
We need to reach consensus on this, and then harmonize the Noctua editing tools and update the ShEx and curator documentation, accordingly.

**Comment by cmungall** [comment]
yes, definitely a more general issue. The physical entities like anatomy are a bit more justifiable as they have changed states throughout a process needing different IDs - this is is standard in reactome too. But for processes, queries such as *how many activities/genes in this process* are fairly central to GO....

**Comment by vanaukenk** [comment]
For the CC and anatomy nodes, I'm mainly thinking about visualization.

@cmungall are you proposing that in anatomy ontologies we'd have terms like, e.g. resting state neuron, depolarized neuron, hyperpolarized neuron and annotate MFs and BPs occurring in these different types of cell state?  How granular do you think we'd need to get wrt cell state?

**Comment by cmungall** [comment]
No, sorry, I didn't mean that. My main point was that sharing instances for physical entities is a bit more tricky. A better example might be a cell component that is transported from one place to another - what is the location of the shared instance here? You can see something analogous with Reactome where under the hood they create different instances for different molecules in different states.

But my main point is that I think there is a real use case for sharing BP nodes, and I think this will be more efficient for curators when making larger models. Imagine a model with two MAPK cascades that is part of a larger multicellular process

**Comment by thomaspd** [comment]
Creating separate instances of BP was discussed and accepted as the standard.
